authorizer/notification_endpoint.go

```diff
@@ -0,0 +1,161 @@
+package authorizer
+
+import (
+	"context"
+
+	"github.com/influxdata/influxdb"
+)
+
+var _ influxdb.NotificationEndpointService = (*NotificationEndpointService)(nil)
+
+// NotificationEndpointService wraps a influxdb.NotificationEndpointService and authorizes actions
+// against it appropriately.
+type NotificationEndpointService struct {
+	s influxdb.NotificationEndpointService
+	influxdb.UserResourceMappingService
+	influxdb.OrganizationService
+	influxdb.SecretService
+}
+
+// NewNotificationEndpointService constructs an instance of an authorizing notification endpoint serivce.
+func NewNotificationEndpointService(
+	s influxdb.NotificationEndpointService,
+	urm influxdb.UserResourceMappingService,
+	org influxdb.OrganizationService,
+	srt influxdb.SecretService,
+) *NotificationEndpointService {
+	return &NotificationEndpointService{
+		s:                          s,
+		UserResourceMappingService: urm,
+		OrganizationService:        org,
+		SecretService:              srt,
+	}
+}
+
+func newNotificationEndpointPermission(a influxdb.Action, orgID, id influxdb.ID) (*influxdb.Permission, error) {
+	return influxdb.NewPermissionAtID(id, a, influxdb.NotificationEndpointResourceType, orgID)
```

Comment:
I think @desa suggested that we piggypack off of Organizations' permissions, to avoid having to introduce new types (looks like we introduced new permission types for checks and notification rules as well).

If we wanted to do that, this would be `influxdb.NewPermissionAtID(id, a, influxdb.OrgsResourceType, orgID)` instead.

I'd strongly prefer to do ^ because it's easier and less of a security risk, but if we do want to introduce new permission types, we'll need to:

1. Write a migration script to add these types to existing superadmin users.
2. Add these new types to the Lookup Service.